emcc.py

```diff
@@ -1281,6 +1282,18 @@ def check(input_file):
       shared.Settings.DECLARE_ASM_MODULE_EXPORTS = 1
       logger.warning('Enabling -s DECLARE_ASM_MODULE_EXPORTS=1, since MODULARIZE currently requires declaring asm.js/wasm module exports in full')
 
+    if shared.Settings.MINIMAL_RUNTIME:
+      # Minimal runtime uses a different default shell file
+      if options.shell_path == shared.path_from_root('src', 'shell.html'):
+        options.shell_path = shared.path_from_root('src', 'shell_minimal_runtime.html')
+
+      # Remove the default exported functions 'memcpy', 'memset', 'malloc', 'free', etc. - those should only be linked in if used
+      shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE = []
+
+      # In asm.js always use memory init file to get the best code size, other modes are not supported.
+      if not shared.Settings.WASM:
+        options.memory_init_file = True
```

Comment:
The code size suboptimal memory initializer modes are not supported (if someone would like to integrate those back in, would be ok as long as it occurs in a way that does not require Closure to clean up bloat caused by them)